regression/scripts/slipstream/propeller_interactions.py

```diff
@@ -0,0 +1,311 @@
+    propeller_wake              = compute_propeller_wake(prop, grid_settings, grid_points, plot_velocities=plot_flag)
+    
+    # run the downstream propeller in the presence of this nonuniform flow
+    T, Q, P, Cp, outputs , etap = run_downstream_propeller(prop, propeller_wake, conditions, plot_performance=plot_flag)
+    
+    T_true, Q_true, P_true, Cp_true, etap_true = 13.02920345,0.6556547,308.96999913,0.86969297,0.37703176
+    assert(abs(T-T_true)<1e-6)
+    assert(abs(Q-Q_true)<1e-6)
+    assert(abs(P-P_true)<1e-6)
+    assert(abs(Cp-Cp_true)<1e-6)
+    assert(abs(etap-etap_true)<1e-6)    
+    
+    # Display plots:
+    if plot_flag:
+        plt.show()
+    
+    return
+
+def run_downstream_propeller(prop, propeller_wake, conditions, plot_performance=False):
+    # assess the inflow at the propeller
+    prop_copy = copy.deepcopy(prop)
+    prop_copy.nonuniform_freestream = True
+    prop_copy.origin = np.array([prop.origin[1] ])# only concerned with the impact the upstream prop has on this one
+    prop_copy.rotation = [prop.rotation[1]]
+    prop = compute_propeller_nonuniform_inflow(prop_copy, propeller_wake, conditions)
+    
+    # run the propeller in this nonuniform flow
+    T, Q, P, Cp, outputs , etap = prop.spin(conditions)
+    
+    if plot_performance:
+        plot_propeller_disc_performance(prop,outputs)
+        
+    return T, Q, P, Cp, outputs , etap
+
+def compute_propeller_wake(prop,grid_settings,grid_points, plot_velocities=True):
+    
+    x_plane = prop.origin[1,0] #second propeller, x-location
+    
+    # generate the propeller wake distribution for the upstream propeller
+    prop_copy = copy.deepcopy(prop)
+    prop_copy.origin = np.array([prop.origin[0]])
+    prop_copy.rotation = [prop.rotation[0]]
+    VD                       = Data()
+    cpts                     = 1 # only testing one condition
+    number_of_wake_timesteps = 80
+    init_timestep_offset     = 0
+    time                     = 2.55
+    
+    WD, dt, ts, B, Nr  = generate_propeller_wake_distribution(prop_copy,cpts,VD,init_timestep_offset, time, number_of_wake_timesteps )
+    
+    # compute the wake induced velocities:
+    VD.YC   = grid_points.ymesh
+    VD.ZC   = grid_points.zmesh
+    VD.XC   = x_plane*np.ones_like(VD.YC)
+    VD.n_cp = np.size(VD.YC)
+    V_ind   = compute_wake_induced_velocity(WD, VD, cpts)
+    u       = V_ind[0,:,0]
+    v       = V_ind[0,:,1]
+    w       = V_ind[0,:,2]
+    
+    propeller_wake = Data()
+    propeller_wake.u_velocities  = u
+    propeller_wake.v_velocities  = v
+    propeller_wake.w_velocities  = w
+    propeller_wake.VD            = VD
+    
+    if plot_velocities:
+        # plot the velocities input to downstream propeller
+        plot_propeller_disc_inflow(prop,propeller_wake,grid_points)
+        
+    
+    return propeller_wake
+
+def generate_propeller_grid(prop, grid_settings, plot_grid=True):
```

Comment:
Is this something that should be moved into methods?

Reply:
Yes, good catch

Reply:
Cool. If there's anything else in that script that can be a method, please move it too